flight/Modules/Buzzer/buzzer.c

```diff
@@ -0,0 +1,260 @@
+// Private types
+
+// Private variables
+static bool module_enabled = false;
+static struct pios_thread *buzzerTaskHandle;
+static int8_t buzzerPin = -1; //Pin for buzzer
+static int8_t accChan = -1; //Accessory Channel
+static bool buzzer_settings_updated;
+static bool wasArmed;
+static bool wasDisarmed;
+static 	FlightBuzzerSettingsData buzzerSettings;
+
+// ****************
+// Private functions
+static void buzzerTask(void * parameters);
+static void getTune(uint8_t arming, uint32_t *newTune);
+
+static int32_t BuzzerStart(void)
+{
+	if (module_enabled) {
+		//FlightBuzzerSettingsConnectCallbackCtx(UAVObjCbSetFlag, &buzzer_settings_updated);
+
+		// Start tasks
+		buzzerTaskHandle = PIOS_Thread_Create(buzzerTask, "buzzerBridge", STACK_SIZE_BYTES, NULL, TASK_PRIORITY);
+		TaskMonitorAdd(TASKINFO_RUNNING_BUZZER, buzzerTaskHandle);
+		return 0;
+	}
+	return -1;
+}
+/**
+ * Initialise the module, called on startup
+ * \returns 0 on success or -1 if initialisation failed
+ */
+int32_t BuzzerInitialize(void)
+{
+#ifdef MODULE_Buzzer_BUILTIN
+	module_enabled = true;
+#else
+	uint8_t module_state[MODULESETTINGS_ADMINSTATE_NUMELEM];
+	ModuleSettingsAdminStateGet(module_state);
+	if (module_state[MODULESETTINGS_ADMINSTATE_BUZZER] == MODULESETTINGS_ADMINSTATE_ENABLED) {
+		module_enabled = true;
+	} else {
+		module_enabled = false;
+		return 0;
+	}
+#endif
+	FlightBuzzerSettingsInitialize();
+
+	return 0;
+}
+MODULE_INITCALL(BuzzerInitialize, BuzzerStart);
+
+/**
+ * Main task. It does not return.
+ */
+static void buzzerTask(void * parameters)
+{
+	
+	buzzer_settings_updated = true;
+	wasArmed = false;
+	wasDisarmed = true;
+
+	// Main task loop
+	uint32_t lastSysTime;
+	lastSysTime = PIOS_Thread_Systime();
+	
+	while (true) {
+		PIOS_Thread_Sleep_Until(&lastSysTime, SAMPLE_PERIOD_MS);
+
+		if (buzzer_settings_updated) {
+			buzzer_settings_updated = false;
+			FlightBuzzerSettingsGet(&buzzerSettings);
+
+			accChan = buzzerSettings.AcessoryChan;
+			buzzerPin = buzzerSettings.BuzzerPin;
+			if (buzzerPin == 0)
+				buzzerPin = -1;
+		}
+
+
+		uint8_t arming;
+		FlightStatusArmedGet(&arming);
+		
+		if ((arming == FLIGHTSTATUS_ARMED_ARMED) && wasDisarmed)
+			wasArmed = true;
+
+		if ((arming == FLIGHTSTATUS_ARMED_DISARMED) && wasArmed)
+			wasDisarmed = true;
+
+
+		static uint32_t currTune = 0;
+		static uint32_t currTuneState = 0;
+		uint32_t newTune = 0;
+
+		//get a new tune
+		getTune(arming, &newTune);
+
+		// Do we need to change tune?
+		if ((newTune != 0) && (currTuneState == 0)) {
+			currTune = newTune;
+			currTuneState = currTune;
+		}
+
+
+		// Play tune
+		bool buzzOn = false;
+
+		if (currTune) {
+			buzzOn = (currTuneState&1);  // Buzz when the LS bit is 1
+
+			// Go to next bit in alarm_seq_state
+			currTuneState >>= 1;
+		}
+		
+		if (buzzOn)
+			PIOS_GPIO_Off(buzzerPin);  //needs a inversion check for npn/pnp transistor
+		else
+			PIOS_GPIO_On(buzzerPin);
+
+	}
+}
+
+
+static void getTune(uint8_t arming, uint32_t *newTune)
```

Comment:
Why not just return a `uint32_t`? The pointer usage is correct but returning an uint makes the code simpler IMO.